parsec/core/types/backend_address.py

```diff
@@ -27,58 +27,55 @@
 ]
 
 
-class BackendOrganizationAddrField(fields.Field):
+class BackendOrganizationAddrField(fields.Field[BackendOrganizationAddr]):
     def _deserialize(self, value: object, attr: str, data: object) -> BackendOrganizationAddr:
+        if not isinstance(value, str):
+            raise ValidationError(f"expected 'str' for got '{type(value)}'")
         try:
-            if not isinstance(value, str):
-                raise ValidationError(f"expected 'str' for got '{type(value)}'")
-
             return BackendOrganizationAddr.from_url(value)
         except ValueError as exc:
             raise ValidationError(str(exc)) from exc
 
     def _serialize(
-        self, value: Optional[BackendOrganizationAddr], attr: str, data: object
-    ) -> Optional[str]:
+        self, value: BackendOrganizationAddr | None, attr: str, data: object
+    ) -> str | None:
         if value is None:
             return None
-
+        assert isinstance(value, BackendOrganizationAddr)
```

Comment:
The idea is this: the typing should protect us against this, except if the call to `_serialize` comes from inside `marshmallow`, which is not typed. So the `isinstance`  is there for checking at runtime during the tests (and maybe to prevent the serialization of erroneous data in production)

Reply:
Note: this is done in all `_serialize` method for behavior consistency.